sway-ast/src/literal.rs

```diff
@@ -42,6 +42,7 @@ pub enum LitIntType {
     I16,
     I32,
     I64,
+    B256,
```

Comment:
There is some extra logic right above this with a boolean for b256 values that ought to be pulled out now that `b256` no longer has its `LitInt::ty_opt` set to `None`.

We should probably consider removing the `Option` if there are no valid uses of the `ty_opt` field that don't have a valid `LitIntType` but it's likely we still want to be able to parse literals that are mistyped (for error handling purposes).